cmd/build.go

```diff
@@ -123,6 +123,9 @@ func runBuild(cmd *cobra.Command, _ []string, newClient ClientFactory) (err erro
 		f.Builder = config.Builder
 	}
 	if config.BuilderImage != "" {
+		if f.BuilderImages == nil {
+			f.BuilderImages = make(map[string]string, 1)
+		}
```

Comment:
IMHO this should be in the constructor of `config`.

Reply:
But config is preparing config struct, not modifying func struct.

Also why would I modify it when I don't have to?
It can be nil (and therefore omitted from `func.yaml`) until we need assign to it.

Reply:
or you mean `NewFunction()` ?